### PR TITLE
[core] Improve release process docs

### PR DESCRIPTION
**Pull request by JCQuintas**
- [x] Ensure script pushes a tag instead of branch
- [x] Added announce merge freeze
- [x] Added how to create changelog
- [x] Added instruction in case publish fails
- [ ] Instruction on which package should be updated vs not. Codemod? Package without changes?
- [ ] Note about announcement, is it optional? When do we send and announcement?

### CODE REVIEW

File: scripts/README.md [hunk 1]
```diff
@@ -66,6 +67,8 @@ Once deployed, it will be accessible at https://material-ui-x.netlify.app/ for t
 
 After documentation is deployed, publish a new release on [GitHub releases page](https://github.com/mui/mui-x/releases)
 
+Create a new release on the newly published tag, then copy/paste the new changes in the changelog while removing the "version" and "date" sections. Eg: the first line of the release should be something in the lines of `We'd like to offer a big thanks to...`
```
Comment (LukasTy) [inline scripts/README.md:70]: WDYT about making it shorter and less reliant on existing changelog formatting? 🤔 
```suggestion
Create a new release on the newly published tag, then copy/paste the new changes in the changelog while removing the "version" and "date" sections. If in doubt, check the previous release notes.
```

File: scripts/releaseTag.mjs [hunk 2]
```diff
@@ -67,7 +67,7 @@ async function main(argv) {
     );
   }
 
-  await exec(['git', 'push', muiXRemote.name, tag].join(' '));
+  await exec(['git', 'push', muiXRemote.name, `refs/tags/${tag}`].join(' '));
```
Comment (LukasTy) [inline scripts/releaseTag.mjs:70]: WDYT about going with a shorthand instead? 🤔 
https://stackoverflow.com/a/49906565/3303436
IMHO, it's arguably even more readable. 🙈 
```suggestion
  await exec(['git', 'push', muiXRemote.name, 'tag', tag].join(' '));
```

File: scripts/README.md [hunk 3]
```diff
@@ -93,3 +96,7 @@ You can use the following script in your browser console on any GitHub page to a
 ### Manually create the release tag
 
 If the `pnpm release:tag` fails you can create and push the tag using the following command: `git tag -a v4.0.0-alpha.30 -m "Version 4.0.0-alpha.30" && git push upstream --tag`.
+
+### release:publish failed
+
+If you receive an error message like `There are no new packages that should be published`. Ensure that you are publishing to the correct registry, not `verdaccio` or anything else. If you need to reset your configuration, you can run `npm config delete registry`
```
Comment (LukasTy) [inline scripts/README.md:102]:
```suggestion
If you receive an error message like `There are no new packages that should be published`. Ensure you are publishing to the correct registry, not `verdaccio` or anything else. If you need to reset your configuration, you can run `npm config delete registry`.
```

**Review by LukasTy — COMMENTED** [review]
Nice improvement! 👍 
Thank you for taking care of it. 💯 💙 
> Note about announcement, is it optional? When do we send and announcement?

I would agree that we should at least mark this section as optional.
Lately, it's mostly up to @noraleonte and @joserodolfofreitas to decide when do we want an announcement and it's generation is no longer taken care by the `card-generator`. 🤔

**Review by michelengelen — APPROVED** [review]

**Comment by michelengelen** [comment]
with the changes suggested by @LukasTy that is! ^^

**Comment by JCQuintas** [comment]
I'll be merging this and setup a reminder to talk about the open points in the future, thanks for the input

**Pull request merged**